### PR TITLE
Update guide

### PR DESCRIPTION
**Pull request by wuminzhe**

### CODE REVIEW

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md [hunk 1]
```diff
@@ -80,13 +110,18 @@ contract ExecuteDemo {
...
-Deploy it on the Pangoro Smart Chain. 
-
-Make sure that the endpoint is correct, it must be the ToPangolinEndpoint's address.
-
 Make sure that the latest spec version of pangolin is correct. You can get it from https://pangolin.subscan.io/runtime.
 
+```bash
+npx hardhat flatten ./Caller.sol > ~/Caller.sol
+```
+Copy the flattened Caller.sol to Remix and deploy it to Pangoro Smart Chain. Make sure that the endpoint is correct when deploying , it must be the PangoroToPangolinEndpoint's address.
```
Comment (boundless-forest) [inline docs/sdk/guides/03-execute-remote-smart-contract-call.md]: `Caller.sol`

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md [hunk 2]
```diff
@@ -80,13 +110,18 @@ contract ExecuteDemo {
...
+
 ## Run
 
-1. You can get a estimated fee by calling [the `fee` function](../api-reference#fee) of the ToPangolinEndpoint contract.
-2. Call the `remoteAdd(callee)` with a value as the market fee. The value should greater than or equal to the estimated fee.
\ No newline at end of file
+1. You can get a estimated market fee by calling the [`fee`](../api-reference#fee) function of the PangoroToPangolinEndpoint contract.
+2. Call the `remoteAdd(callee)`. 
+
+   * The param `callee` is the PangolinToPangoroEndpoint contract address.
```
Comment (boundless-forest) [inline docs/sdk/guides/03-execute-remote-smart-contract-call.md]: The `callee` here refers to the `PangolinToPangoroEndpoint` contract or the `Callee.sol` deployed contract?

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md [hunk 3]
```diff
@@ -46,21 +76,21 @@ contract Callee {
 }
 ```
 
-Deploy it on the Pangolin Smart Chain.
+Copy it to Remix and deploy it to Pangolin Smart Chain.
 
 ## Create the caller contract on Pangoro
```
Comment (boundless-forest) [inline docs/sdk/guides/03-execute-remote-smart-contract-call.md]:
```suggestion
## Deploy the caller contract on the Pangoro Smart Chain
```

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md [hunk 4]
```diff
@@ -8,29 +8,59 @@ This Dapp allows you to call Pangolin Smart Chain's `add(2)` function from Pango
...
-3. Call the `setRemoteEndpoint(_remoteChainId, _remoteEndpoint)` of the two endpoints to point to each other. 
+4. Call the `setRemoteEndpoint(_remoteChainId, _remoteEndpoint)` of the two endpoints to point to each other. 
 
-The chain id of Pangoro Smart Chain is `0x70616772`. The chain id of Pangolin Smart Chain is `0x7061676c`.  
+   The chain id of Pangoro Smart Chain is `0x70616772`. The chain id of Pangolin Smart Chain is `0x7061676c`.  
 
-You can get the full chain id list from [constants](../constants).
+   > You can get the full chain id list from [constants](../constants).
 
 ## Create the callee contract on Pangolin
```
Comment (boundless-forest) [inline docs/sdk/guides/03-execute-remote-smart-contract-call.md]:
```suggestion
## Deploy the callee contract on the Pangolin Smart Chain
```

File: docs/sdk/guides/04-dispatch-remote-substrate-call.md [hunk 5]
```diff
@@ -18,56 +18,9 @@ npm install --save-dev @darwinia/contracts-periphery @darwinia/contracts-utils
...
-                pangolinCallWeight
-            );
-    }
-}
-```
-
-Deploy it on the Pangoro Smart Chain. The `remoteDispatch` will be used in the next step.
-
-You can download the completed [ToPangolinEndpoint.sol](https://raw.githubusercontent.com/darwinia-network/darwinia-messages-sol/master/contracts/periphery/contracts/s2s/examples/ToPangolinEndpoint.sol), and add your own access controls to it if your want to use it in a production environment.
+Download [PangoroToPangolinEndpoint.sol](https://github.com/darwinia-network/darwinia-s2s-template/blob/main/contracts/PangoroToPangolinEndpoint.sol) for Pangoro.  Deploy it on the Pangoro Smart Chain. The `remoteDispatch` will be used in the next step.
```
Comment (boundless-forest) [inline docs/sdk/guides/04-dispatch-remote-substrate-call.md]: Why not provide a wget download command here?

File: docs/sdk/guides/04-dispatch-remote-substrate-call.md [hunk 6]
```diff
@@ -121,5 +74,8 @@ Deploy the Dapp contract on the Pangoro Smart Chain. Inject the endpoint address
 
 ## Run
 
-1. You can get a estimated fee by calling [the `fee` function](../api-reference#fee) of the endpoint contract.
-2. Call the `remoteRemark(_remark)` with a value as the market fee. The value should greater than or equal to the estimated fee.
+1. You can get a estimated market fee by calling [the `fee` function](../api-reference#fee) of the endpoint contract.
+2. Call the `remoteRemark(_remark)`.
+
+   * The param `_remark` is the content you want to remark.  
+   * The value input should be set to a market fee >= the estimated market fee.
```
Comment (boundless-forest) [inline docs/sdk/guides/04-dispatch-remote-substrate-call.md:92]:
```suggestion
   * The value input should be set to a market fee(Get in the first step). 
```

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md [hunk 7]
```diff
@@ -80,13 +110,18 @@ contract ExecuteDemo {
...
 ## Run
 
-1. You can get a estimated fee by calling [the `fee` function](../api-reference#fee) of the ToPangolinEndpoint contract.
-2. Call the `remoteAdd(callee)` with a value as the market fee. The value should greater than or equal to the estimated fee.
\ No newline at end of file
+1. You can get a estimated market fee by calling the [`fee`](../api-reference#fee) function of the PangoroToPangolinEndpoint contract.
+2. Call the `remoteAdd(callee)`. 
+
+   * The param `callee` is the PangolinToPangoroEndpoint contract address.   
+   * The value input should be set to a market fee >= the estimated market fee.
```
Comment (boundless-forest) [inline docs/sdk/guides/03-execute-remote-smart-contract-call.md:126]:
```suggestion
  * The value input should be set to a market fee(Get in the first step). 
```

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md (hunk 2 above)
Comment (wuminzhe) [reply docs/sdk/guides/03-execute-remote-smart-contract-call.md]: You are right. The callee does not refer to the PangolinToPangoroEndpoint contract

File: docs/sdk/guides/04-dispatch-remote-substrate-call.md (hunk 5 above)
Comment (wuminzhe) [reply docs/sdk/guides/04-dispatch-remote-substrate-call.md]: right

File: docs/sdk/guides/04-dispatch-remote-substrate-call.md (hunk 6 above)
Comment (wuminzhe) [reply docs/sdk/guides/04-dispatch-remote-substrate-call.md:92]: fee > estimated market fee is allowed.

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md (hunk 7 above)
Comment (wuminzhe) [reply docs/sdk/guides/03-execute-remote-smart-contract-call.md:126]: same

File: docs/sdk/guides/03-execute-remote-smart-contract-call.md (hunk 1 above)
Comment (wuminzhe) [reply docs/sdk/guides/03-execute-remote-smart-contract-call.md]: the endpoint param is PangoroToPangolinEndpoint's address

**Review by boundless-forest — APPROVED** [review]

**Pull request merged**